site/en/docs/extensions/mv3/tut_websockets/index.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: 'layouts/doc-post.njk'
+title: 'Using WebSockets in Service Workers'
+seoTitle: 'How to use WebSockets in Chrome Extension Service Workers'
+date: 2023-07-04
+description: 'Step-by-step instructions on how to connect to a WebSocket in your Chrome Extension.'
+---
+
+This tutorial demonstrates how to connect to a WebSocket in your Chrome Extension's service worker. You can find a [working example on Github](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/tutorial.websockets).
+
+## Background
+
+Starting with Chrome M116, extension service workers get full support for WebSockets. Previously, a service worker could become inactive if the WebSocket connection was active for more than 30s and no other extension events occurred at that time, which could result in connection loss and the inability to receive messages.
+
+The reason for this is that Chrome terminates an extension service worker after 30s of inactivity. In Chrome versions &lt;M116 messages sent or received on a WebSocket do not extend the service worker lifetime, causing Chrome to suspend the service workers with an active WebSocket connection. For more background on the extension service worker lifecycle, read the [extension service worker guide](/docs/extensions/mv3/service_workers/service-worker-lifecycle/#idle-shutdown)).
```

Comment:
you already mentioned the 30 seconds in the previous sentence.

Reply:
"causing Chrome to suspend the service workers with an active WebSocket connection"

It doesn't _cause_ it. The order of the sentence should be flipped I think. i.e. Chrome <m116 would suspend connections because messages wouldn't reset the timer.

That being said, I think this sentence more or less just repeats the previous sentence, it can probably be combined or removed.

Reply:
good point, I really struggled to write that paragraph. I'm using Oliver's suggestion.